### PR TITLE
Xdn 10520 add source element to rum cls event reports mark brocato

### PR DESCRIPTION
**Pull request by markbrocato**

### CODE REVIEW

File: src/getSelectorForElement.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/**
+ * Returns an array of element descriptors for the element and each of it's parent's going up to the document element.
```
Comment (ierceg) [inline src/getSelectorForElement.ts:2]:
```suggestion
 * Returns an array of element descriptors for the element and each of its parent's going up to the document element.
```

File: src/Metrics.ts [hunk 2]
```diff
@@ -81,21 +82,36 @@ declare var Metrics: MetricsConstructor
...
-  private metrics: { [name: string]: number | undefined } = {}
+  private metrics: { [name: string]: number | string | undefined }
   private token?: string
   private options: MetricsOptions
   private sendTo: string
   private originalURL: string
   private pageID: string
   private index: number = 0
   private clientNavigationHasOccurred: boolean = false
+  private xdnEnvironmentID?: string
```
Comment (ierceg) [inline src/Metrics.ts:93]:
```suggestion
  private layer0EnvironmentId?: string
```

?

**Comment by markbrocato** [comment]
@ierceg Comments addressed.  Please review again.

File: src/Metrics.ts [hunk 3]
```diff
@@ -221,13 +267,36 @@ class BrowserMetrics implements Metrics {
...
+          entry.route &&
+          new RegExp(entry.route, 'i').test(location.pathname)
+      )
+      return matchingRoute?.criteriaPath
+    }
   }
 
   /**
-   * Sends all collected metrics to Layer0 RUM.
+   * Sends all collected metrics to Moovweb RUM.
```
Comment (markbrocato) [inline src/Metrics.ts:299]:
```suggestion
   * Sends all collected metrics to Layer0 RUM.
```

**Review by ierceg — APPROVED** [review]

**Pull request merged**